File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/updater/parts/GetPartInfoMessage.scala





















// Generated on 05/03/2013 10:35:01
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GetPartInfoMessage extends MessageDeserializer {
  val messageId = 1506

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GetPartInfoMessage.messageId

    val id: String

    def serialize(buf: Buffer) {
      buf.writeString(id)
    }

  }

  protected def create(id: String): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val id = buf.readString()

    create(id)
  }

}

object GetPartInfoMessage extends GetPartInfoMessage {
  case class MessageImpl(id: String) extends Message
  type Target = MessageImpl

  override def create(id: String) = MessageImpl(id)

  def apply(id: String) = create(id)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GetPartInfoMessage.java


// Generated on 05/08/2013 19:37:59
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GetPartInfoMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 1506;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public String id;
    
    public GetPartInfoMessage() { }
    
    public GetPartInfoMessage(String id) {
        this.id = id;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeString(id);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        id = buf.readString();
    }
    
}
